### PR TITLE
Query: Adds TRIM string system function support in LINQ

### LINKED ISSUE #3781: Support Trim in LINQ
**Is your feature request related to a problem? Please describe.**
The SDK throw when using Trim in Linq.
```text
Microsoft.Azure.Cosmos.Linq.DocumentQueryException: Method 'Trim' is not supported.
ActivityId: 01c9c3ff-9834-4fa1-93f3-8f112b94fabd, Windows/10.0.22621 cosmos-netstandard-sdk/3.30.2
   at Microsoft.Azure.Cosmos.Linq.StringBuiltinFunctions.Visit(MethodCallExpression methodCallExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.BuiltinFunctionVisitor.VisitBuiltinFunctionCall(MethodCallExpression methodCallExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitNonSubqueryScalarExpression(Expression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitNonSubqueryScalarExpression(Expression expression, ReadOnlyCollection`1 parameters, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitScalarExpression(Expression expression, ReadOnlyCollection`1 parameters, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitScalarExpression(Expression expression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitBinary(BinaryExpression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitNonSubqueryScalarExpression(Expression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitNonSubqueryScalarExpression(Expression expression, ReadOnlyCollection`1 parameters, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitScalarExpression(Expression expression, ReadOnlyCollection`1 parameters, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitScalarExpression(Expression expression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitBinary(BinaryExpression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitNonSubqueryScalarExpression(Expression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitNonSubqueryScalarExpression(Expression expression, ReadOnlyCollection`1 parameters, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitScalarExpression(Expression expression, ReadOnlyCollection`1 parameters, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitScalarExpression(LambdaExpression lambda, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitWhere(ReadOnlyCollection`1 arguments, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitMethodCall(MethodCallExpression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.Translate(Expression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.TranslateQuery(Expression inputExpression, IDictionary`2 parameters, CosmosLinqSerializerOptions linqSerializerOptions)
   at Microsoft.Azure.Cosmos.Linq.SqlTranslator.TranslateQuery(Expression inputExpression, CosmosLinqSerializerOptions linqSerializerOptions, IDictionary`2 parameters)
   at Microsoft.Azure.Cosmos.Linq.DocumentQueryEvaluator.HandleMethodCallExpression(MethodCallExpression expression, IDictionary`2 parameters, CosmosLinqSerializerOptions linqSerializerOptions)
   at Microsoft.Azure.Cosmos.Linq.DocumentQueryEvaluator.Evaluate(Expression expression, CosmosLinqSerializerOptions linqSerializerOptions, IDictionary`2 parameters)
   at Microsoft.Azure.Cosmos.Linq.CosmosLinqQuery`1.CreateFeedIterator(Boolean isContinuationExpected)
   at Microsoft.Azure.Cosmos.Linq.CosmosLinqQuery`1.ToFeedIterator()
   at Microsoft.Azure.Cosmos.Linq.CosmosLinqExtensions.ToFeedIterator[T](IQueryable`1 query)
```

**Describe the solution you'd like**
Trim function is supported in LINQ

**Describe alternatives you've considered**
Trim works with `QueryDefinition`

**Additional context**
I'm using Microsoft.AspNetCore.OData.Query to receive an OData query from the user and apply it to CosmosDB Linq provider. For now, I'm stuck since OData only support Trim, while current version of the SDK only support LTRIM and RTRIM.

### PR DESCRIPTION
**Pull request by leminh98**
# Pull Request Template

## Description

This PR adds support to convert Linq Trim() function call to Sql's TRIM system function

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #3781

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs [hunk 1]
```diff
@@ -348,6 +348,45 @@ protected override SqlScalarExpression VisitImplicit(MethodCallExpression method
             }
         }
 
+        private class StringVisitTrim : SqlBuiltinFunctionVisitor
+        {
+            public StringVisitTrim()
+                : base("TRIM",
+                    false,
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:355]: It doesn't seem like this parameter is used.

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs [hunk 2]
```diff
@@ -348,6 +348,45 @@ protected override SqlScalarExpression VisitImplicit(MethodCallExpression method
...
+                    char[] argumentsExpressions = (char[])((ConstantExpression)methodCallExpression.Arguments[0]).Value;
+                    if (argumentsExpressions.Length == 0)
+                    {
+                        validInNet = true;
+                    }
+                }
+                else if (methodCallExpression.Arguments.Count == 0)
+                {
+                    validInNetCore = true;
+                }
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:378]: This code seems to be common across many visitors.
Consider adding a common base class that has this logic and simply reuse.

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs [hunk 3]
```diff
@@ -348,6 +348,45 @@ protected override SqlScalarExpression VisitImplicit(MethodCallExpression method
...
+                bool validInNetCore = false;
+
+                if (methodCallExpression.Arguments.Count == 1 &&
+                    methodCallExpression.Arguments[0].NodeType == ExpressionType.Constant &&
+                    methodCallExpression.Arguments[0].Type == typeof(char[]))
+                {
+                    char[] argumentsExpressions = (char[])((ConstantExpression)methodCallExpression.Arguments[0]).Value;
+                    if (argumentsExpressions.Length == 0)
+                    {
+                        validInNet = true;
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:372]: Curious, do we have coverage for this case?

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs [hunk 4]
```diff
@@ -348,6 +348,45 @@ protected override SqlScalarExpression VisitImplicit(MethodCallExpression method
...
+                    }
+                }
+                else if (methodCallExpression.Arguments.Count == 0)
+                {
+                    validInNetCore = true;
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:377]: Do you have coverage for this case?

Comment (leminh98) [reply Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:377]: Yes, this is just a .Trim() call

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs (hunk 2 above)
Comment (leminh98) [reply Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:378]: I think that should stay in a separate PR

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs (hunk 1 above)
Comment (leminh98) [reply Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:355]: Not really sure when the implicit is being used, but we are keeping them like this for consistency across visitors

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs (hunk 3 above)
Comment (leminh98) [reply Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:372]: added

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs (hunk 1 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:355]: I did a quick search and it's not used even in the base class. A cleanup may be in order. I am ok if you want to do it separately.

**Pull request merged**